drivers/SmartThings/sonos/src/utils.lua

```diff
@@ -122,6 +122,33 @@ function utils.new_mac_address_keyed_table()
   return setmetatable({}, _mac_addr_key_mt)
 end
 
+---@param tbl table<string,any>
+---@param key string
+local function __case_insensitive_key_index(tbl, key)
+  assert(type(key) == "string", "key for CaseInsensitiveKeyTable must be a string!")
+  local raw = rawget(tbl, key)
+  if raw ~= nil then
+    return raw
+  end
+  local lowercase = key:lower()
+  return rawget(tbl, lowercase)
+end
+
+local function __case_insensitive_key_newindex(tbl, key, value)
+  assert(type(key) == "string", "key for CaseInsensitiveKeyTable must be a string!")
+  rawset(tbl, key:lower(), value)
+end
+
+local _case_insensitive_key_mt = {
+  __index = __case_insensitive_key_index,
+  __newindex = __case_insensitive_key_newindex,
+  __metatable = "CaseInsensitiveKeyTable",
```

Comment:
What does setting a metatable to a string do?

Reply:
Setting `__metatable` to anything non-nil creates a "protected" metatable that can't be overridden by calling `setmetatable`, and it will return that value when `getmetatable` is used.

So it's kinda like setting a `.class` name/value if you use a string instead of a table.